lib/util/host-rules.ts

```diff
@@ -28,6 +30,18 @@ export function add(params: HostRule) {
     throw new Error('hostRules cannot contain both a hostName and baseUrl');
   }
   hostRules.push(params);
+  const confidentialFields = ['password', 'token'];
+  confidentialFields.forEach(field => {
+    const secret = params[field];
+    if (secret && secret.length > 3 && !secrets.includes(secret))
+      secrets.push(secret);
+  });
+  if (params.username && params.password) {
+    const secret = Buffer.from(
```

Comment:
We should move this to a util function, because i've seen this on some more places.